packages/snowpack/src/logger.ts

```diff
@@ -1,35 +1,97 @@
+class SnowpackLogger {
+  /** set the log level (can be changed after init) */
+  public level: LoggerLevel = 'info';
+  /** configure maximum number of logs to keep (default: 500) */
+  public logCount = 500;
 
-    return msg;
+  private history: string[] = []; // this is immutable; must be accessed with Logger.getHistory()
+  private callbacks: Record<LoggerEvent, (message: string) => void> = {
+    debug: (message: string) => {
+      console.log(message);
+    },
+    info: (message: string) => {
+      console.log(message);
+    },
+    warn: (message: string) => {
+      console.warn(message);
+    },
+    error: (message: string) => {
+      console.error(message);
+    },
   };
-}
 
-/** use Pino, a tiny logging library */
-export default function createLogger(opts: pino.LoggerOptions = {}) {
-  return pino({
-    ...opts,
-    name: opts.name || 'snowpack',
-    prettyPrint: {suppressFlushSyncWarning: true},
-    prettifier,
-  });
+  private log({level, name, message}: {level: LoggerEvent; name: string; message: string}) {
+    // test if this level is enabled or not
+    if (levels[this.level] > levels[level]) {
+      return; // do nothing
+    }
+
+    // format
+    let text = message;
+    if (level === 'warn') text = colors.yellow(text);
+    if (level === 'error') text = colors.red(text);
+    const log = `${colors.dim(`[${name}]`)} ${text}`;
+
+    // add to log history and remove old logs to keep memory low
+    this.history.push(log);
+    while (this.history.length > this.logCount) {
+      this.history.shift();
+    }
+
+    // log
+    if (typeof this.callbacks[level] === 'function') {
+      this.callbacks[level](log);
+    } else {
+      throw new Error(`No logging method defined for ${level}`);
+    }
+  }
+
+  /** emit messages only visible when --debug is passed */
+  public debug(message: string, options?: LoggerOptions): void {
+    const name = (options && options.name) || 'snowpack';
+    this.log({level: 'debug', name, message});
+  }
+
+  /** emit general info */
+  public info(message: string, options?: LoggerOptions): void {
+    const name = (options && options.name) || 'snowpack';
+    this.log({level: 'info', name, message});
+  }
+
+  /** emit non-fatal warnings */
+  public warn(message: string, options?: LoggerOptions): void {
+    const name = (options && options.name) || 'snowpack';
+    this.log({level: 'warn', name, message});
+  }
+
+  /** emit critical error messages */
+  public error(message: string, options?: LoggerOptions): void {
+    const name = (options && options.name) || 'snowpack';
+    this.log({level: 'error', name, message});
+  }
+
+  /** get full logging history */
+  public getHistory() {
+    return [...this.history];
+  }
+
+  /** listen for events */
+  public on(event: LoggerEvent, callback: (message: string) => void) {
+    this.callbacks[event] = callback;
+  }
 }
+
+/** export one logger to rest of app */
+export default new SnowpackLogger();
```

Comment:
nit: since we're a closed codebase, lets use named exports. It makes find-replace & refactoring easier when the use of this is forced to always be the same name (vs. different files importing via different variable names)